crates/blas-tests/tests/oper.rs

```diff
@@ -280,7 +280,7 @@ fn gen_mat_mul()
                         cv = c.view_mut();
                     }
 
-                    let answer_part = alpha * reference_mat_mul(&av, &bv) + beta * &cv;
+                    let answer_part: Array<f64, _> = alpha * reference_mat_mul(&av, &bv) + beta * &cv;
```

Comment:
Is this change actually needed?

Reply:
Let me double check this later, it might not be. In terms of breaking changes, this is one kinda-uncontrollable issue: Rust's type inference may run into some issues. But perhaps this is unnecessary, or otherwise something we can fix.

Reply:
Ah fascinating; yes, it does need it. If you don't include it, you get the following compiler error:
```shell
error[E0271]: type mismatch resolving `<ArrayBase<OwnedRepr<f64>, Dim<[usize; 2]>> as Add>::Output == ArrayRef<f64, _>`
   --> crates/blas-tests/tests/oper.rs:283:75
    |
283 |                     let answer_part = alpha * reference_mat_mul(&av, &bv) + beta * &cv;
    |                                                                           ^ expected `ArrayBase<OwnedRepr<f64>, Dim<[usize; 2]>>`, found `ArrayRef<f64, _>`
    |
    = note: expected struct `ndarray::ArrayBase<OwnedRepr<f64>, Dim<[usize; 2]>>`
               found struct `ArrayRef<f64, _>`
```
which I was initially confused by. The key is that the following line:
```rust
answer.slice_mut(s![..;s1, ..;s2]).assign(&answer_part);
```
calls `.assign`, which now expects `&ArrayRef<_, _>`. So the compiler tries to infer the type of `answer_part` as `ArrayRef<_, _>`, which isn't possible. Fascinating look into Rust's type inference. I feel like it should be able to infer the `Array` type since it knows that both both `cv` and the output of `reference_mat_mul` are both `Array`, but I guess not.